pkg/connector/tickets.go

```diff
@@ -0,0 +1,461 @@
+package connector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strconv"
+	"time"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	sdkTicket "github.com/conductorone/baton-sdk/pkg/types/ticket"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	jira "github.com/conductorone/go-jira/v2/cloud"
+)
+
+type TicketManager interface {
+	GetTicket(ctx context.Context, ticketId string) (*v2.Ticket, annotations.Annotations, error)
+	CreateTicket(ctx context.Context, ticket *v2.Ticket, schema *v2.TicketSchema) (*v2.Ticket, annotations.Annotations, error)
+	GetTicketSchema(ctx context.Context, schemaID string) (*v2.TicketSchema, annotations.Annotations, error)
+	ListTicketSchemas(ctx context.Context, pToken *pagination.Token) ([]*v2.TicketSchema, string, annotations.Annotations, error)
+}
+
+func getJiraStatusesForProject(ctx context.Context, client *jira.Client, projectId string) ([]jira.JiraStatus, error) {
+	var jiraStatuses []jira.JiraStatus
+	statusOffset := 0
+	statusMaxResults := 100
+
+	for {
+		// Fetch statuses here and pass in to schemaForProject
+		statuses, resp, err := client.Status.SearchStatusesPaginated(ctx,
+			jira.WithStartAt(statusOffset),
+			jira.WithMaxResults(statusOffset+statusMaxResults),
+			jira.WithStatusCategory("DONE"),
+			jira.WithProjectId(projectId))
+		if err != nil {
+			return nil, err
+		}
+
+		jiraStatuses = append(jiraStatuses, statuses...)
+
+		totalGot := statusOffset + resp.MaxResults
+		if totalGot >= resp.Total {
+			break
+		}
+		statusOffset += resp.Total
```

Comment:
`resp.Total` is the total number of projects. we should be adding `resp.MaxResults` to the `statusOffset` then we can check if `statusOffset > resp.Total` and break

Reply:
ohhh gotcha

Reply:
https://github.com/ConductorOne/baton-jira/pull/13/commits/d63fcc895707f3ef60334c0a376c358db14465b4